File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val = 0, _next = nil)
#         @val = val
#         @next = _next
#     end
# end
# @param {ListNode} head
# @return {ListNode}
def delete_duplicates(head)
    return head if head == nil
    curr, curr_val = head, head.val
    while curr
        while curr.next && curr.next.val == curr_val
            curr.next = curr.next.next
        end
        curr = curr.next
        if curr
            curr_val = curr.val
        end
    end
    return head
end

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *   Val int
 *   Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
  if head == nil {
    return head
  }
  curr, currVal := head, head.Val
  for curr != nil {
    for curr.Next != nil && curr.Next.Val == currVal {
      curr.Next = curr.Next.Next
    }
    curr = curr.Next
    if curr != nil {
      currVal = curr.Val
    }
  }
  return head
}